File: Answer/3315.构造最小位运算数组-ii.go
/*
 * @lc app=leetcode.cn id=3315 lang=golang
 *
 * [3315] 构造最小位运算数组 II
 */

// @lc code=start
func minBitwiseArray(nums []int) []int {
	n := len(nums)
	ans := make([]int, n)
	for i := 0; i < n; i++ {
		value := nums[i]
		if value == 2 {
			ans[i] = -1
		} else {
			t := ^value
			lowbit := t & -t
			ans[i] = value ^ (lowbit >> 1)
		}
	}
	return ans
}

// @lc code=end

File: Answer/3315.构造最小位运算数组-ii.java
/*
 * @lc app=leetcode.cn id=3315 lang=java
 *
 * [3315] 构造最小位运算数组 II
 */

// @lc code=start
import java.util.List;

class Solution {

    public int[] minBitwiseArray(List<Integer> nums) {
        var n = nums.size();
        var ans = new int[n];
        for (int i = 0; i < n; i++) {
            var value = nums.get(i);
            if (value == 2) {
                ans[i] = -1;
            } else {
                int t = ~value;
                int lowbit = t & -t;
                ans[i] = value ^ (lowbit >> 1);
            }
        }
        return ans;
    }
}
// @lc code=end

